internal/build/phase_config_provider.go

```diff
@@ -122,14 +123,38 @@ func WithBinds(binds ...string) PhaseConfigProviderOperation {
 	}
 }
 
-func WithDaemonAccess() PhaseConfigProviderOperation {
+func WithDaemonAccess(dockerHost string) PhaseConfigProviderOperation {
 	return func(provider *PhaseConfigProvider) {
 		WithRoot()(provider)
-		bind := "/var/run/docker.sock:/var/run/docker.sock"
-		if provider.os == "windows" {
-			bind = `\\.\pipe\docker_engine:\\.\pipe\docker_engine`
+		if dockerHost == "inherit" {
+			dh, ok := os.LookupEnv("DOCKER_HOST")
```

Comment:
Is there any case we're getting by using `LookupEnv` here (and the ensuing if/else), as opposed to just doing something like:
```
dockerHost = os.GetEnv("DOCKER_HOST")
```
?

It seems like whether someone set it to `""`, or it wasn't set, we're going to assign dockerHost to `""`, and then it will be given the default values in the next for loops, so we may as well simplify it.